services/codecov/codecov.service.js

```diff
@@ -100,40 +99,33 @@ module.exports = class Codecov extends BaseJsonService {
     }
   }
 
-  async fetch({ vcsName, user, repo, branch, token }) {
-    // Codecov Docs: https://docs.codecov.io/reference#section-get-a-single-repository
-    let url = `https://codecov.io/api/${vcsName}/${user}/${repo}`
-    if (branch) {
-      url += `/branches/${branch}`
-    }
-    const options = {}
-    if (token) {
-      options.headers = {
-        Authorization: `token ${token}`,
-      }
-    }
-    return this._requestJson({
+  async fetch({ vcsName, user, repo, branch, token, flag }) {
+    const url = `https://codecov.io/${vcsName}/${user}/${repo}${
+      branch ? `/branches/${branch}` : ''
+    }/graph/badge.svg`
+    const queryParams = [flag && `flag=${flag}`, token && `token=${token}`]
+      .filter(Boolean)
+      .join('&')
+    return this._requestSvg({
       schema,
-      options,
-      url,
-      errorMessages: {
-        401: 'not authorized to access repository',
-        404: 'repository not found',
-      },
```

Comment:
Is this because Codecov will always return a 200 response, but with an `unknown` in the case of an invalid repository? 

If so, it might be better to throw a `NotFound` error in the case of `unknown` for the user experience. A lot of times folks will have a minor typo in one of the route params (i.e. `/badges/sheilds/`) and it's helpful if we can provide a more meaningful badge message in those cases.

Reply:
> Is this because Codecov will always return a 200 response, but with an unknown in the case of an invalid repository?

Yes. Their badge returns `unknown` when:
- repository doesn't exist
- coverage is yet not uploaded for a repo
- coverage for a specific flag doesn't exist or hasn't been uploaded yet

Reply:
Ah I see. That's a very minor downside in the svg scraping approach since we no longer have the ability to differentiate. We should probably leave this as you have it here :+1: